src/wallet/signer.rs

```diff
@@ -234,6 +256,10 @@ impl Signer for DescriptorXKey<ExtendedPrivKey> {
         false
     }
 
+    fn id(&self, secp: &SecpCtx) -> SignerId {
```

Comment:
Yeah I wasn't really thinking about it because our internal context that we keep inside `Wallet` is an `All` ctx, so it doesn't really matter. But that's a good point, for the public-facing APIs we should use the right traits to avoid restricting it too much

Reply:
Ah wait, it looks like it can't be done because it would require declaring a generic type parameter on `id()`, which then makes it impossible to use `Singer` as a trait-object.

Reply:
ahhhh of course! Not much we can do here then. It might be interesting to find out if `Wallet` can work with a `Signing` only instance but I wouldn't want to add another type parameter for Wallet at this stage.

Reply:
I think we could do it without adding an extra type parameter because we create the context inside `Wallet` (which I think is reasonable considering that it's not something you are supposed to do many times repeatedly), so we could even create it as a `SignOnly`.

But I think we have some stuff that require the `Verification` ctx, iirc even things like computing the fingerprint of an xpub wanted that flag, so we are forced to keep it as `All` unfortunately